src/main/java/com/commercetools/sync/producttypes/ProductTypeSync.java

```diff
@@ -240,30 +261,66 @@ private void handleError(@Nonnull final String errorMessage, @Nullable final Thr
      * @param newProductType draft containing data that could differ from data in {@code oldProductType}.
      * @return a {@link CompletionStage} which contains an empty result after execution of the update.
      */
-    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
     @Nonnull
-    private CompletionStage<Void> updateProductType(@Nonnull final ProductType oldProductType,
-                                                    @Nonnull final ProductTypeDraft newProductType) {
-
-        final List<UpdateAction<ProductType>> updateActions = buildActions(oldProductType, newProductType, syncOptions);
-
-        final List<UpdateAction<ProductType>> updateActionsAfterCallback = syncOptions.applyBeforeUpdateCallBack(
-            updateActions,
-            newProductType,
-            oldProductType
-        );
-
-        if (!updateActionsAfterCallback.isEmpty()) {
-            return productTypeService.updateProductType(oldProductType, updateActionsAfterCallback)
-                    .thenAccept(updatedProductType -> statistics.incrementUpdated())
-                    .exceptionally(exception -> {
-                        final String errorMessage = format(CTP_PRODUCT_TYPE_UPDATE_FAILED, newProductType.getKey());
-                        handleError(errorMessage, exception, 1);
+    private CompletionStage<Optional<ProductType>> updateProductType(
+        @Nonnull final ProductType oldProductType,
+        @Nonnull final ProductTypeDraft newProductType,
+        @Nonnull final List<UpdateAction<ProductType>> updateActions) {
+
+        return productTypeService
+            .updateProductType(oldProductType, updateActions)
+            .handle(ImmutablePair::new)
+            .thenCompose(updateResponse -> {
+                final ProductType updatedProductType = updateResponse.getKey();
+                final Throwable sphereException = updateResponse.getValue();
```

Comment:
I think we don`t use the term "sphere" anymore...maybe rename it to ctException

Reply:
@ahalberkamp actually the exception type is exactly called `SphereException`: http://commercetools.github.io/commercetools-jvm-sdk/apidocs/io/sphere/sdk/models/SphereException.html